### ISSUE TITLE
zstd: provide a "contains magic number" func

### ISSUE THREAD

**Issue by mvdan** [klauspost/compress#253]
That is, to detect if a chunk of bytes has the zstd magic number as a prefix.

There are Go modules like https://github.com/h2non/filetype, which support lots of file types. But at the end of the day, I just need to check four bytes, and it seems unnecessary to add an extra module dependency when I already depend on a zstd package :)

For example, here's a sample implementation I wrote, which could be used like `if zstd.HasMagic(chunk) { ...`:

```
// HasMagic reports whether src begins with zstd v1's magic number,
// 0xFD2FB528 in little-endian format.
func HasMagic(src []byte) bool {
        return len(src) >= 4 &&
                src[0] == 0x28 && src[1] == 0xB5 &&
                src[2] == 0x2f && src[3] == 0xFD
}
```

I only raised this as an issue instead of a PR because the patch is tiny and I feel like there's a lot of room for bikeshedding, or discussion around whether or not we want this.

**Comment by mvdan** [comment]
(But of course, happy to submit a PR if this seems like a good idea)

**Comment by klauspost** [comment]
It could be part of #237 which would also check more frame header fields.

**Comment by mvdan** [comment]
Is that from the point of view of exposing less API, or so that the function does extra sanity checks?

I can sympathize with exposing less API. I don't care about extra sanity checks though, as I directly follow up "is the magic number there" with "decompress the bytes". I guess I could ignore the rest of the data as long as it's equally cheap to retrieve.

**Comment by klauspost** [comment]
Less API 💯

**Comment by mvdan** [comment]
Sounds good to me :) I'll continue using a local helper while I wait for this issue to be fixed as part of #237 - there's no rush.

**Comment by chrislusf** [comment]
Any update on this?

It would be great if the magic number can be prefixed without API changes.
